localstack/services/s3/virtual_host.py

```diff
@@ -9,20 +9,19 @@
 from localstack.runtime import hooks
 from localstack.services.edge import ROUTER
 from localstack.services.s3.utils import S3_VIRTUAL_HOST_FORWARDED_HEADER
-from localstack.utils.aws.request_context import AWS_REGION_REGEX
 
 LOG = logging.getLogger(__name__)
 
+AWS_REGION_REGEX = r"(?:us-gov|us|ap|ca|cn|eu|sa)-[a-z]+-\d"
+
 # virtual-host style: https://{bucket-name}.s3.{region?}.{domain}:{port?}/{key-name}
 # ex: https://{bucket-name}.s3.{region}.localhost.localstack.cloud.com:4566/{key-name}
 # ex: https://{bucket-name}.s3.{region}.amazonaws.com/{key-name}
-VHOST_REGEX_PATTERN = f"<regex('.*'):bucket>.s3.<regex('({AWS_REGION_REGEX}\\.)?'):region><regex('.*'):domain><regex('(?::\\d+)?'):port>"
+VHOST_REGEX_PATTERN = f"<regex('.*'):bucket>.s3.<regex('(?:{AWS_REGION_REGEX}.)?'):region><domain>"
```

Comment:
Did you remove the escaping of the `.` on purpose (i.e. change the semantics from "region followed by a ." to "region followed by any character")?

Reply:
whoopsie. no i didn't, good catch!